broker/artemis/plugin/src/main/java/org/eclipse/kapua/broker/artemis/plugin/security/context/SecurityContextHandler.java

```diff
@@ -141,17 +276,55 @@ public Subject buildFromPrincipal(KapuaPrincipal kapuaPrincipal) {
         return subject;
     }
 
-    public void printContent(String method, String connectionId) {
-        StringBuilder builder = new StringBuilder();
-        builder.append("\n============================").append(method).append(" - ").append(connectionId);
-        builder.append("\nconnection info by client id\n");
-        sessionContextMapByClient.forEach((key, sessionContext) -> builder.append("\tclientId: ").append(key).append(" - conId: ").append(sessionContext.getConnectionId()).append("\tinternal: ").append(sessionContext.isInternal()).append("\n"));
-        builder.append("\nconnection info by connection id\n");
-        sessionContextMapByConnection.forEach((key, sessionContext) -> builder.append("\tconId: ").append(key).append(" - clientId: ").append(sessionContext.getClientId()).append("\tinternal: ").append(sessionContext.isInternal()).append("\n"));
-        builder.append("\nacl by connection id\n");
-        aclMap.forEach((key, acl) -> builder.append("\tconnId: ").append(key).append("\n"));
-        builder.append("\nprincipal by connection id\n");
-        principalMap.forEach((key, principal) -> builder.append("\tconnId: ").append(key).append(" - name: ").append(principal.getName()).append(" - clientId: ").append(principal.getClientId()).append("\tinternal: ").append(principal.isInternal()).append("\n"));
-        logger.info("{}", builder);
+    public void updateStealingLinkAndIllegalState(AuthRequest authRequest, String connectionId, String oldConnectionId) {
+        authRequest.setStealingLink(isStealingLink(connectionId, oldConnectionId));
+        authRequest.setIllegalState(isIllegalState(authRequest));
+    }
+
+    private boolean isStealingLink(String connectionId, String oldConnectionId) {
+        return oldConnectionId!=null && connectionId!=null ?
+            !connectionId.equals(oldConnectionId) : oldConnectionId!=null;
```

Comment:
It shouldn't be equivalent because if oldConnectionId is null there is no stealing link (the client had no previous connection established) but, with the suggested changes, the resutl is true (!someString.equals(null) => true)
Then if oldConnectionId is null the method should return false, do you agrre?

Reply:
Okay you're right, I see the difference now.  So in your code though if `oldConnectionId="someString" and connectionId=null` then it would return true, which I assume is what you want?

What do you think of this version:
```java
return oldConnectionId != null && !oldConnectionId.equals(connectionId)
```

So if oldConnectionId is null or oldConnectionId is equal to connectionId then it returns false.  Otherwise if oldConnectionId is not null and it doesn't equal connectionId (including when connectionId is null) then it returns true.

Reply:
sounds good even thought the condition `connectionI==null` shouldn't occour